UI and spacing improvements to list view

devinreams opened the pull request:
Resolves #157 

This should address all the items in the issue ☝️ above

<img width="273" alt="screen shot 2018-04-02 at 12 49 39 pm" src="https://user-images.githubusercontent.com/49511/38210418-971b523e-3674-11e8-83cc-00fb9bd275d2.png">

- [x] I read some articles and played around with the way the separators are inserted. The simplest thing to get the desired effect was make it so the footer item does _not_ have a border (by setting a height), adjust the insets on the cells, and remove the inset on the overall table (so we get a full width border separating the filter). How does everything look to you @nickbrandt?
- [x] I don't really know my way around interface builder but I think I got these addressed as simply/concisely/cleanly as possible. Requesting @sashei review and confirm and make sure any other code stuff I'd need to do is done.

nickbrandt commented:
@devinreams From what I can see it looks good. Can't seem to get the build to install on my phone but the screenshot looks correct (can't really see borders but from your comments that sounds right). Thanks for knocking these out!

devinreams commented:
>For instances where you are filtering and there are no results, the border between the filter and list area is lost, and replaced with a white line. Ideally this would remain the gray filter border color 

Good catch @nickbrandt. I updated the footer view color so it's not white:

<img width="470" alt="screen shot 2018-04-03 at 5 18 35 am" src="https://user-images.githubusercontent.com/49511/38246290-8bff765c-36fe-11e8-97b2-054c1d8c61d5.png">

<img width="483" alt="screen shot 2018-04-03 at 5 18 39 am" src="https://user-images.githubusercontent.com/49511/38246291-8c3a6d02-36fe-11e8-80c7-bb4f11c6bebc.png">


Want to take one final look at this before we call it "done"?

And I see @changecourse opened an issue for an improved (non-empty) empty state: #186
